[TaskProcessing] Use taskprocessing in TextProcessing and STT managers

julien-nc opened the pull request:
When
* scheduling a text processing task
* running a text processing task
* scheduling a STT task
* running a STT task

a task processing task actually runs. So these old APIs still benefit from the new task processing providers.

@marcelklehr Like we discussed, it might be good to remove the AI admin settings STT section and adjust the text processing one so it does not show preferences for `FreePromptTaskType`, `HeadlineTaskType`, `SummaryTaskType` and `TopicsTaskType`.

Also, @marcelklehr Should we only run the task processing task in those manager's "run" methods and get rid of the 2nd part which still runs the task like before in case the task processing task has failed?

julien-nc committed:
feat(textprocessing): TextProcessingManager::runTask calls TaskProcessingManager::runTask

Signed-off-by: Julien Veyssier <julien-nc@posteo.net>

julien-nc committed:
feat(speech-to-text): SpeechToTextManager::transcribeFile calls TaskProcessingManager::runTask

Signed-off-by: Julien Veyssier <julien-nc@posteo.net>

julien-nc commented:
/backport to stable30

julien-nc committed:
fix(taskprocessing): fix tests

Signed-off-by: Julien Veyssier <julien-nc@posteo.net>

bigcat88 reviewed (approved):
looks good

marcelklehr commented on lib/private/SpeechToText/SpeechToTextManager.php:
```diff
@@ -112,7 +116,28 @@ public function cancelScheduledFileTranscription(File $file, ?string $userId, st
+			}
+		} catch (Throwable $e) {
+			$this->logger->debug('Failed to run a Speech-to-text job from STTManager with TaskProcessing for file ' . $file->getId(), ['exception' => $e]);
+		}
```
@julien-nc What do you think about adding a bit more logic here to

1. Avoid calling task processing if audio2text is not available
2. Properly pass on the failure if it is available and fails

instead of a generic fall back

marcelklehr commented on lib/private/TaskProcessing/Manager.php:
```diff
@@ -98,8 +96,34 @@ public function __construct(
+		}
+
+		return $providers;
+	}
```
Instead of duplicating the logic, what do you think about having a `getTextProcessingManager()` method that summons the manager to avoid the circular dependency?

marcelklehr committed:
fix(TaskProcessing): Use OCP\Server::get instead of copying methods

Signed-off-by: Marcel Klehr <mklehr@gmx.net>
Signed-off-by: Julien Veyssier <julien-nc@posteo.net>

marcelklehr committed:
fix(SpeechToTextManager): Throw TaskProcessing Task failed

Signed-off-by: Marcel Klehr <mklehr@gmx.net>
Signed-off-by: Julien Veyssier <julien-nc@posteo.net>

marcelklehr committed:
fix(settings): Remove STT admin settings

taskprocessing is transparent to STT providers so specific STT settings are obsolete

Signed-off-by: Marcel Klehr <mklehr@gmx.net>
Signed-off-by: Julien Veyssier <julien-nc@posteo.net>

marcelklehr committed:
fix(settings): Don't show built-in textprocessing task types in textprocessing settings

TaskProcessing is transparent to textprocessing providers and TextProcessing can use Taskprocessing providers so these are unnecessary

Signed-off-by: Marcel Klehr <mklehr@gmx.net>
Signed-off-by: Julien Veyssier <julien-nc@posteo.net>

marcelklehr committed:
fix(TextProcessing): Don't run taskProcessing if task type is not available

Signed-off-by: Marcel Klehr <mklehr@gmx.net>
Signed-off-by: Julien Veyssier <julien-nc@posteo.net>

marcelklehr committed:
fix(TaskProcessing): Fix namespace of TaskProcessingTest.php

Signed-off-by: Marcel Klehr <mklehr@gmx.net>
Signed-off-by: Julien Veyssier <julien-nc@posteo.net>

marcelklehr committed:
Revert "fix(TaskProcessing): Use OCP\Server::get instead of copying methods"

This reverts commit d624c8da8f0dc2eaf18d416aa15db0a2035de398.

Signed-off-by: Julien Veyssier <julien-nc@posteo.net>

marcelklehr commented:
/compile

nextcloud-command committed:
chore(assets): Recompile assets

Signed-off-by: nextcloud-command <nextcloud-command@users.noreply.github.com>
